akka-stream/src/main/scala/akka/stream/serialization/StreamRefSerializer.scala

```diff
@@ -0,0 +1,187 @@
+    case ref: SourceRefImpl[_]                       ⇒ serializeSourceRef(ref).toByteArray
+    //    case ref: MaterializedSourceRef[_]               ⇒ serializeSourceRef(ref.).toByteArray
+  }
+
+  override def fromBinary(bytes: Array[Byte], manifest: String): AnyRef = manifest match {
+    // protocol
+    case OnSubscribeHandshakeManifest ⇒ deserializeOnSubscribeHandshake(bytes)
+    case SequencedOnNextManifest      ⇒ deserializeSequencedOnNext(bytes)
+    case CumulativeDemandManifest     ⇒ deserializeCumulativeDemand(bytes)
+    case RemoteSinkCompletedManifest  ⇒ deserializeRemoteStreamCompleted(bytes)
+    case RemoteSinkFailureManifest    ⇒ deserializeRemoteStreamFailure(bytes)
+    // refs
+    case SinkRefManifest              ⇒ deserializeSinkRef(bytes)
+    case SourceRefManifest            ⇒ deserializeSourceRef(bytes)
+  }
+
+  // -----
+
+  private def serializeCumulativeDemand(d: StreamRefsProtocol.CumulativeDemand): StreamRefMessages.CumulativeDemand = {
+    StreamRefMessages.CumulativeDemand.newBuilder()
+      .setSeqNr(d.seqNr)
+      .build()
+  }
+
+  private def serializeRemoteSinkFailure(d: StreamRefsProtocol.RemoteStreamFailure): StreamRefMessages.RemoteStreamFailure = {
+    StreamRefMessages.RemoteStreamFailure.newBuilder()
+      .setCause(ByteString.copyFrom(d.msg.getBytes))
+      .build()
+  }
+
+  private def serializeRemoteSinkCompleted(d: StreamRefsProtocol.RemoteStreamCompleted): StreamRefMessages.RemoteStreamCompleted = {
+    StreamRefMessages.RemoteStreamCompleted.newBuilder()
+      .setSeqNr(d.seqNr)
+      .build()
+  }
+
+  private def serializeOnSubscribeHandshake(o: StreamRefsProtocol.OnSubscribeHandshake): StreamRefMessages.OnSubscribeHandshake = {
+    StreamRefMessages.OnSubscribeHandshake.newBuilder()
+      .setTargetRef(StreamRefMessages.ActorRef.newBuilder()
+        .setPath(Serialization.serializedActorPath(o.targetRef)))
+      .build()
+  }
+
+  private def serializeSequencedOnNext(o: StreamRefsProtocol.SequencedOnNext[_]) = {
+    val p = o.payload.asInstanceOf[AnyRef]
+    val msgSerializer = serialization.findSerializerFor(p)
+
+    val payloadBuilder = StreamRefMessages.Payload.newBuilder()
+      .setEnclosedMessage(ByteString.copyFrom(msgSerializer.toBinary(p)))
+      .setSerializerId(msgSerializer.identifier)
+
+    msgSerializer match {
+      case ser2: SerializerWithStringManifest ⇒
+        val manifest = ser2.manifest(p)
+        if (manifest != "")
+          payloadBuilder.setMessageManifest(ByteString.copyFromUtf8(manifest))
+      case _ ⇒
+        if (msgSerializer.includeManifest)
+          payloadBuilder.setMessageManifest(ByteString.copyFromUtf8(p.getClass.getName))
+    }
+
+    StreamRefMessages.SequencedOnNext.newBuilder()
+      .setSeqNr(o.seqNr)
+      .setPayload(payloadBuilder.build())
+      .build()
+  }
+
+  private def serializeSinkRef(sink: SinkRefImpl[_]): StreamRefMessages.SinkRef = {
```

Comment:
I think this could just be `Serialization.serializedActorPath(sink.initialPartnerRef)` as utf8 bytes.

Reply:
Instead of protobufing sinkref? Kind of wanted that for forwards compatibility